shell/src/main/java/alluxio/cli/docgen/MetricsDocGenerator.java

```diff
@@ -68,19 +72,26 @@ public static void generate() throws IOException {
       String category = typeStr.toLowerCase();
       metricTypeMap.put(typeStr, category);
     }
-
+    String csvFolder = PathUtils.concatPath(homeDir, CSV_FILE_DIR);
+    String ymlFolder = PathUtils.concatPath(homeDir, YML_FILE_DIR);
     try (Closer closer = Closer.create()) {
       Map<FileWriterKey, FileWriter> fileWriterMap = new HashMap<>();
-      String csvFolder = PathUtils.concatPath(homeDir, CSV_FILE_DIR);
-      String ymlFolder = PathUtils.concatPath(homeDir, YML_FILE_DIR);
       FileWriter csvFileWriter;
       FileWriter ymlFileWriter;
       for (String category : CATEGORIES) {
-        csvFileWriter = new FileWriter(PathUtils
-            .concatPath(csvFolder, category + "-metrics." + CSV_SUFFIX));
-        csvFileWriter.append(CSV_FILE_HEADER + "\n");
-        ymlFileWriter = new FileWriter(PathUtils
-            .concatPath(ymlFolder, category + "-metrics." + YML_SUFFIX));
+        if (validate) {
+          csvFileWriter = new FileWriter(PathUtils
+              .concatPath(csvFolder, ("temp-" + category + "-metrics." + CSV_SUFFIX)));
```

Comment:
similar comment regarding the use of `temp-` string prefix; if it needs to be the same across several places, share the same reference